snippets/csharp/VS_Snippets_CLR_System/system.hashcode.structure/cs/example3.cs

```diff
@@ -0,0 +1,61 @@
+// <Snippet1>
+using System;
+using System.Collections.Generic;
+
+public struct Path : IEquatable<Path>
+{
+    public IReadOnlyList<string> Segments { get; }
+
+    public Path(params string[] segments)
+    {
+        Segments = segments;
+    }
+
+    public override bool Equals(object obj) => obj is Path o && Equals(o);
+
+    public bool Equals(Path other)
+    {
+        if (ReferenceEquals(Segments, other.Segments)) return true;
+        if (Segments is null || other.Segments is null) return false;
+        if (Segments.Count != other.Segments.Count) return false;
+
+        for (var i = 0; i < Segments.Count; i++)
```

Comment:
Also could use `?.

Reply:
This can never be null (due to the 2nd if statement that checks for null inequality).

Reply:
I caught that in most == implementations, but missed this.